### PR TITLE
[frontend][ONNX]support ConvTranspose explicitly specified output_shape

### PR DESCRIPTION
**Pull request by ah-cheng**
When I import the onnx model efficientdet_lite.onnx. There are some ConvTranspose operator just provide `output_shape` attributes. no `pads`. As shown in the picture below:
image
Then i get a error is the
 ```
AssertionError: Output shapes (1, 256, 256, 16) and (1, 270, 270, 16) don't match
```
According the doc: https://github.com/onnx/onnx/blob/main/docs/Operators.md#ConvTranspose
```
total_padding[i] = stride[i] * (input_size[i] - 1) + output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - output_shape[i]
```
I updated how `total_padding` is calculated. And support  explicitly specified the ```output_shape```.

cc: @MatthewARM  @jwfromm  @AndrewZhaoLuo

### CODE REVIEW

**Comment by AndrewZhaoLuo** [comment]
I'll look at this tomorrow

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -2815,6 +2815,48 @@ def repeat(N, D):
 
 @tvm.testing.parametrize_targets
 def test_convtranspose(target, dev):
+    def verify_convtranspose_with_output_shape(
```
Comment (AndrewZhaoLuo) [inline tests/python/frontend/onnx/test_forward.py:2818]: Can you remove "test_convtranspose_output_shape", from `unsupported_onnx_tests`?

File: python/tvm/relay/frontend/onnx.py [hunk 2]
```diff
@@ -688,13 +691,28 @@ def _impl_v11(cls, inputs, attr, params):
...
+                        total_pad[i] = (
+                            output_padding[i]
+                            + ((kernel_shape[i] - 1) * dilations[i] + 1)
+                            - strides[i]
+                        )
                 left = [p // 2 for p in total_pad]
                 right = [total_pad[i] - left[i] for i in range(kndim)]
-                if "LOWER" in attr["auto_pad"]:
+                if "output_shape" in attr and "auto_pad" not in attr:
+                    # assert left[0] == right[0] and left[1] == right[1],\
```
Comment (AndrewZhaoLuo) [inline python/tvm/relay/frontend/onnx.py:712]: remove comment if it's not needed

File: python/tvm/relay/frontend/onnx.py [hunk 3]
```diff
@@ -688,13 +691,28 @@ def _impl_v11(cls, inputs, attr, params):
                 output_padding = attr.get("output_padding", [0] * kndim)
                 strides = attr["strides"]
                 total_pad = [0] * kndim
-                for i in range(kndim):
-                    total_pad[i] = (
-                        output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - strides[i]
-                    )
+                if "output_shape" in attr:
```
Comment (AndrewZhaoLuo) [inline python/tvm/relay/frontend/onnx.py:694]: Can you post a reference back to the docs on how these are calculated? https://github.com/onnx/onnx/blob/main/docs/Operators.md#ConvTranspose

**Review by AndrewZhaoLuo — COMMENTED** [review]
LGTM, just please enable the test

**Comment by AndrewZhaoLuo** [comment]
Looks like we don't match onnx behavior based on official tests.

**Comment by ah-cheng** [comment]
I am confused. Why did the test pass before. And I can also test it pass locally.
Maybe I will check it again today.

**Comment by ah-cheng** [comment]
Looks like it because of   remove "test_convtranspose_output_shape". Right?

**Comment by ah-cheng** [comment]
Looks like it's an onnx official issue.  According to the code：
```
https://github.com/onnx/onnx/blob/8ab1efa4fc979e1748ebd378a71a59a22deabe88/onnx/defs/nn/defs.cc
```
the function `convTransposeShapeInference` we can know. 
It did not perform additional processing on the value of pad after explicitly specifying output_shape.
So according to the test
https://github.com/onnx/onnx/blob/main/onnx/backend/test/case/node/convtranspose.py
The default pads is [0,0,0,0]
So the last result is [1, 2, 9, 7]

So about this PR. I should close it now until the onnx office fix it?  Or I still add the `test_convtranspose_output_shape` in `unsupported_onnx_tests`?  So you can review this PR?

**Comment by AndrewZhaoLuo** [comment]
We should match behavior in the official onnx test. Don't have time to take a look deeply at this, but if you can figure it out it will be cool B)

To run the failing test you can do

`pytest tests/python/frontend/onnx/test_forward.py::test_onnx_nodes[test_convtranspose_output_shape-llvm]` 

(You can get the failing test names from the Jenkins "test" tab)

**Comment by ah-cheng** [comment]
OK, thank you very much for your reply. @AndrewZhaoLuo .  I will continue to follow up on this issue.
BTW, I have raised an issue on the official onnx to discuss this issue. When necessary, I will also raise PR to fix this problem, and then we will follow up with this PR.

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM, looks like you jsut got a flaky test though, please just push an empty commit to restart ci

**Comment by ah-cheng** [comment]
It seems not auto merge, Please help to review again. Thanks. 
I found the version of official onnx test `test_convtranspose_output_shape`  is `version 9`, It is the reason why test fail at begin. now, I fix it and it test success.

**Comment by AndrewZhaoLuo** [comment]
Thanks! Was out on vacation last week so apologies for not merging sooner

**Pull request merged**